harness/async-gc.js

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2019  Ecma International.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+description: >
+    Collection of functions used to capture references cleanup from garbage collectors
+features: [Symbol, async-functions]
+flags: [non-deterministic]
+features: [FinalizationGroup]
+---*/
+
+function asyncGC(...targets) {
+  var fg = new FinalizationGroup(() => {});
+  var length = targets.length;
+
+  for (let target of targets) {
+    fg.register(target, 'target');
+    target = null;
+  }
+
+  targets = null;
+
+  return Promise.resolve('tick').then(() => asyncGCDeref()).then(() => {
+    var names;
+
+    // consume iterator to capture names
+    fg.cleanupSome(iter => { names = [...iter]; });
+
+    if (!names || names.length != length) {
+      throw asyncGC.notCollected;
+    }
+  });
+}
+
+asyncGC.notCollected = Symbol('Object was not collected');
+
+async function asyncGCDeref() {
+  var trigger;
+
+  // TODO: Remove this when $262.clearKeptObject becomes documented and required
+  if ($262.clearKeptObjects) {
+    trigger = $262.clearKeptObjects();
+  }
+
+  await $262.gc();
+
+  return Promise.resolve(trigger);
+}
+
+function resolveAsyncGC(err) {
+  // if (err === asyncGC.notCollected) {
+  //   // Do not fail as GC can't provide necessary resources.
+  //   $DONE();
+  // }
+
+  $DONE(err);
```

Comment:
Am I correct in understanding that `resolveAsyncGC` passes along `asyncGC.notCollected` and report executions that didn't collect all the targets as FAILED, and it'll be up to implementations to interpret that as okay because of the `non-deterministic` flag?

Reply:
Sorry, I didn’t mean to commit The commented lines, the intention is to call $DONE without args, which will end with a passing state. Thanks for catching this.